server/routes/reddit.js

```diff
@@ -9,26 +8,33 @@ const r = new snoowrap({
   refreshToken: process.env.REDDIT_REFRESH_TOKEN
 });
 
-router.get("/search/new/:company", (req, res, next) => {
-  r.search({query: req.params.company, sort: 'new'})
+const getNewestRedditPosts = async (company) => {
+  let submissions = [];
+  await r.search({query: company, sort: 'new'})
     .then(posts => {
-      let submissions = [];
       posts.forEach((submission, i) => {
-        console.log(submission.title);
-        console.log(submission.subreddit_name_prefixed);
-        console.log(submission.permalink);
-        console.log(submission.thumbnail);
-        console.log(submission.selftext);
-        submissions.push({
-          title: submission.title,
-          platform: submission.subreddit_name_prefixed,
-          link: submission.permalink,
-          image: submission.thumbnail,
-          desc: submission.selftext
-        });
+        if (submission.thumbnail === "image") {
+          submissions.push({
+            title: submission.title,
+            platform: submission.subreddit_name_prefixed,
+            link: "https://www.reddit.com" + submission.permalink,
+            image: submission.url,
+            desc: submission.selftext
+          });
+        } else {
+          submissions.push({
+            title: submission.title,
+            platform: submission.subreddit_name_prefixed,
+            link: submission.url,
+            image: (submission.thumbnail !== "default" && submission.thumbnail !== "self") ? submission.thumbnail : "https://a.thumbs.redditmedia.com/9EDGp3AsLDtCRvDUAjuQzNQSZPkvVmgesMjVxphosb0.jpg",
+            desc: submission.selftext
+          });
+        }
```

Comment:
Can probably handle the logic for setting the image in a variable outside of the submissions.push(...) and then stick that into the push instead of making two nearly identical pushes. 
`let img;`
`if(submission.thumbnail === 'image'){
   img = submission.url;
}`
`else {
   img = (submission.thumbnail !== "default" && submission.thumbnail !== "self") ? submission.thumbnail : "https://a.thumbs.redditmedia.com/9EDGp3AsLDtCRvDUAjuQzNQSZPkvVmgesMjVxphosb0.jpg"
}`

`submissions.push({
            title: submission.title,
            platform: submission.subreddit_name_prefixed,
            link: "https://www.reddit.com" + submission.permalink,
            image: submission.url,
            desc: submission.selftext
          });`

Reply:
Alright I changed it